students/ScottL/session04/Mailroom_Part2.py

```diff
@@ -0,0 +1,185 @@
+                        {"Name": "Tom Petty", "Amount": 15000},
+                        {"Name": "Neil Diamond", "Amount": 20000},
+                        {"Name": "Neil Diamond", "Amount": 30000},
+                        {"Name": "Elton John", "Amount": 25000},
+                        {"Name": "Elton John", "Amount": 25000},
+                        {"Name": "Elton John", "Amount": 40000},
+                        {"Name": "John Lennon", "Amount": 10000},
+                        {"Name": "Mick Jagger", "Amount": 20000},
+                        {"Name": "Mick Jagger", "Amount": 10000}]
+MENU_HEADER_1 = "\nMenu Options:"
+MENU_HEADER_2 = "\n" + "-" * len(MENU_HEADER_1)
+MENU_SELECTIONS = {1: "Send a 'Thank You' note to individual donor",
+                   2: "Print a report",
+                   3: "Send 'Thank You' letters to all donors",
+                   4: "Quit"}
+REPORT_HEADER_1 = "\n{:<26}{:<12}{:^17}{:>11}".format(
+                    "Donor Name", "Total Dollars", "Number", "Avg Amount")
+REPORT_HEADER_2 = "\n" + "-" * len(REPORT_HEADER_1)
+DONOR_LIST_HEADER = "\nDonor Name:\n" + "-" * 20
+
+
+def display_menu_selections():
+    """Display the contents of the menu options for the user."""
+    print(MENU_HEADER_1, MENU_HEADER_2)
+    for item in MENU_SELECTIONS:
+        print(item, ": ", MENU_SELECTIONS[item])
+
+
+def get_user_input():
+    """Prompt user to make a selection from the menu and return back to Main Menu."""
+    return input("\nSelect a number from the menu options (1-4): ")
+
+
+def calc_total_donations_per_donor():
+    """Return a list of donors with their total donations in descending order."""
+    sorted_donation_list = []
+    for item in global_donation_list:
+        bool_flag = False
+        for entry in sorted_donation_list:
+            if entry["Name"] == item["Name"]:
+                entry["Amount"] += item["Amount"]
+                bool_flag = True
+        if bool_flag is False:
+            sorted_donation_list.append({"Name": item["Name"], "Amount": item["Amount"]})
+    sorted_donation_list.sort(key=lambda x: x["Amount"], reverse=True)
+    return sorted_donation_list
+
+
+def count_num_of_donations_per_donor():
+    """Return a dictionary of donors and their respective number of donations."""
+    number_of_donations = {}
+    for item in global_donation_list:
+        if item["Name"] in number_of_donations:
+            number_of_donations[item["Name"]] += 1
+        else:
+            number_of_donations[item["Name"]] = 1
+    return number_of_donations
+
+
+def calc_avg_donation_per_donor(list_of_donations, donation_number):
+    """
+    Return a dictionary of the mean donation size for each donor.
+    arg1 = list containing a string of names in the first element
+            and an integer/float in the second element
+    arg2 = list containing a string of names in the first element
+            and an integer in the second element
+    """
```

Comment:
actual proper syntax for doc strings are as follows:
```
    """
    Return a dictionary of the mean donation size for each donor.
    :param list_of_donations: list containing a string of names in the first element and an integer/float in the second element
    :type list_of_donations: list
    :param donation_number: list containing a string of names in the first element and an integer in the second element
     :type donation_number: int
    """
```
if you update all your functions with this syntax and then in a python interpreter do
```
>>> import Mailroom_Part2
>>> help(Mailroom_Part2)
```
and be amazed.